src/api/RequestApi.ts

```diff
@@ -401,8 +507,88 @@ export type IResponseItem = Omit<
 // This extends the IInRequest to drop some required objects and add additional objects
-export type IRequestItem = Omit<IResponseItem, "supGovLead" | "employee"> & {
+export type IInRequestItem = Omit<
+  IInResponseItem,
+  "supGovLead" | "employee"
+> & {
+  supGovLeadId: number;
+  employeeId: number;
+  employeeStringId?: string;
+};
+
+// create IItem item to work with it internally
+export type IOutRequest = {
+  reqType: REQUEST_TYPES;
+  /** Required - Will be -1 for NewForms that haven't been saved yet */
+  Id: number;
+  /** Required - Contains the Employee's Name */
+  empName: string;
+  /** Required - Employee's Type valid values are:
+   * 'Civilian' - for Civilian Employees
+   * 'Contractor' - for Contracted Employees
+   * 'Military' - for Military Employees
+   */
+  empType: EMPTYPES;
+  /** Optional - The Employee's SAR from the UMD -- Required for CIV/MIL, others will be blank */
+  SAR?: number;
+  /** Optional - The Employee's Sensitivity Code from the PD -- Required for CIV, others will be blank */
+  sensitivityCode?: number;
+  /** Required - The user's last day */
+  lastDay: Date;
+  /** Required - The Expected Completion Date - Default to 28 days from eta*/
+  beginDate: Date;
+  /** Required - The Superviosr/Gov Lead of the employee */
+  supGovLead: IPerson;
+  /** Optional - The employee GAL entry. If the user doesn't exist yet, then it will be undefined */
+  employee?: IPerson;
+  /** Optional - Date Supervisor Closed or Cancelled -- If there is a cancelReason then we know it was cancelled */
+  closedOrCancelledDate?: Date;
+  /** Optional - The reason for why the request was cancelled */
+  cancelReason?: string;
+  // Required - This is a field internally used by the app -- it is calculated within the app and not passed to/from the data repo (SharePoint)
+  status: "Active" | "Cancelled" | "Closed";
+};
+
+// create PnP JS response interface for the InForm
+// This extends the IOutRequest to change the types of certain objects
+export type IOutResponseItem = Omit<
+  IOutRequest,
+  "lastDay" | "beginDate" | "closedOrCancelledDate" | "status" // Drop the status object from the type, as it is used internally and is not data from the repository
+> & {
+  // Storing the date objects in Single Line Text fields as ISOStrings
+  lastDay: string;
+  beginDate: string;
+  closedOrCancelledDate?: string;
+};
+
+// create PnP JS response interface for the InForm
+// This extends the IOutRequest to drop some required objects and add additional objects
+export type IOutRequestItem = Omit<
+  IOutResponseItem,
+  "supGovLead" | "employee"
+> & {
   supGovLeadId: number;
   employeeId: number;
   employeeStringId?: string;
 };
+
+// Custom Type Checking Function to determine if it is an In Processing Request, or an Out Processing Request
+export function isInRequest(
+  request: IInRequest | IOutRequest
+): request is IInRequest {
+  return request.reqType === REQUEST_TYPES.InRequest;
+}
+
```

Comment:
So this is used to help Typescript understand the type of data.  When this is called, it knows if it returns true then the data is of type IInRequest.  If it doesn't then it knows the data will be of type IOutReqest.  This makes other things more simple, because just directly checking the reqType wouldn't help it to know the type of data expected, unless we then do typecasting based on the results.  This wraps it all into 1 nice call.